src/v/cluster/node/local_monitor.cc

```diff
@@ -89,6 +92,30 @@ ss::future<struct statvfs> local_monitor::get_statvfs(const ss::sstring& path) {
     }
 }
 
+float percent_free(const disk& disk) {
+    long double free = disk.free, total = disk.total;
+    return float((free / total) * 100.0);
+}
+
+void local_monitor::maybe_log_space_error(const disk& disk) {
+    auto [min_by_bytes, min_by_percent] = minimum_free_by_bytes_and_percent(
+      disk.total);
+    auto min_space = std::min(min_by_percent, min_by_bytes);
+    clusterlog.log(
+      ss::log_level::error,
+      despam_interval,
+      "{}: free space at {:.3f}% on {}: {} total, {} free, "
+      "min. free {}. Please adjust retention policies as needed to "
+      "avoid running out of space.",
+      stable_alert_string,
+      percent_free(disk),
+      disk.path,
+      // TODO: generalize human::bytes for unsigned long
+      human::bytes(disk.total), // NOLINT narrowing conv.
```

Comment:
nit: you can instruct clang-tidy to be very specific with the syntax `NOLINT(name of rule)`

Reply:
I like this.. for some reason my editor does not. Its using clang-tidy for warnings but the only expression I can get to work is `(*)`. It seems to trigger two similar warnings, `cppguidelines-narrowing-conversions` and `bugprone-narrowing-conversions`. If I use the intuitive match `// NOLINT(*narrowing-conversions)` it only eliminates one of them :shrug: